src/librustc_trait_selection/traits/select/mod.rs

```diff
@@ -1104,6 +1104,14 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         // who might care about this case, like coherence, should use
         // that function).
         if candidates.is_empty() {
+            // If there's an error type, 'downgrade' our result from
+            // `Err(Unimplemented)` to `Ok(None)`. This helps us avoid
+            // emitting additional spurious errors, since we're guaranteed
+            // to have emitted at least one.
+            if stack.obligation.references_error() {
+                debug!("no results for error type, treating as ambiguous");
```

Comment:
Add a `delay_span_bug`, just in case.

Reply:
With https://github.com/rust-lang/rust/pull/70551, seeing an error type guarantees that `delay_span_bug` has been called.

Reply:
r=me when CI is green then :)

Reply:
Did you want to wait for #70551, or can I r+ this now?